pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -159,6 +159,23 @@ func getPVSpecFromCache(name string, pvcReadOnly bool, expectedClaimUID types.UI
 	return volume.NewSpecFromPersistentVolume(clonedPV, pvcReadOnly), nil
 }
 
+// DetermineVolumeAction returns true if volume and pod needs to be added to dswp
+// and it returns false if volume and pod needs to be removed from dswp
+func DetermineVolumeAction(pod *v1.Pod, desiredStateOfWorld cache.DesiredStateOfWorld, defaultValue bool) bool {
```

Comment:
> The volume action (attach or detach) is decided not only by state of the pod (terminated or not) but also if user has asked kubelet to keep terminated volumes around.

IMO, the most important decision is done before calling the function and is passed as `defaultValue`. That's why the name seems misleading to me - `DetermineVolumeAction` implies that whole decision is done here. Anyway, it's just a function name, DetermineVolumeAction will do.